persistence/src/main/scala/za/co/absa/spline/persistence/ArangoImplicits.scala

```diff
@@ -36,6 +36,14 @@ object ArangoImplicits {
         ) yield res.next
       }
 
+      def queryAll[T: Manifest](queryString: String,
```

Comment:
Let's call it `queryStream` or `queryMany` to better denote the purpose. To me `queryAll` reads as _entire collection_, or a greedy fetchin.

Otherwise LGTM.